### ISSUE TITLE
Not working with Mongoose 4.0.1 / Mongo JS 2.0

### ISSUE THREAD

**Issue by buunguyen** [aheckmann/gridfs-stream#72]
Using `fineOne()` of gridfs-stream with Mongoose 4.0 throws the following error.
```
Unhandled rejection Error: collection name must be a String
    at Error (native)
    at checkCollectionName (/app/node_modules/mongoose/node_modules/mongodb/lib/utils.js:70:11)
    at Function.Collection (/app/node_modules/mongoose/node_modules/mongodb/lib/collection.js:57:3)
    at Grid.findOne (/app/node_modules/gridfs-stream/lib/index.js:138:110)
```

It looks like Mongoose uses Mongo JS driver 2.0, which changes the signature of the [`Collection` constructor](https://github.com/mongodb/node-mongodb-native/blob/2.0/lib%2Fcollection.js#L56) where collection name is expected at the 4th position while gridfs-stream passes to the [2nd position](https://github.com/aheckmann/gridfs-stream/blob/master/lib/index.js#L136). I wanted to submit a PR but am just not sure how to deal with the other parameters. Besides, I'm not sure if there's any other change.

**Comment by kjunine** [comment]
+1

**Comment by Reggino** [comment]
Hmmm... ```gridfs-stream``` is  compatible with Monodb 2.0.x and all unit tests seem to succeed... Can you add a test for this?

**Comment by buunguyen** [comment]
I created a bare bone repo here: https://github.com/buunguyen/gridfs-stream-issue. There's a [single script](https://github.com/buunguyen/gridfs-stream-issue/blob/master/app.js) which uses Mongoose 4.0 and gridfs-stream. Upon the call to `findOne`, it throws the error I reported above. (I tried with both Node 0.10 and iojs.)

**Comment by Reggino** [comment]
Thanks. I've added your test to the suite and fixed the issue... This is fixed in ```1.1.1```

**Comment by buunguyen** [comment]
Awesome, thank you!
